rxnorm/rxnorm_extract.py

```diff
@@ -0,0 +1,181 @@
+            # If the concept has a line that is RXNORM with a desired TTY (IN\BN\PIN) we will place the parameters according to this line
+            if row['SAB'] == 'RXNORM' and row['TTY'] in ('BN', 'IN', 'PIN'):
+                bn_in_pin = row['TTY']
+                had_rxnorm = True
+                name = row['STR']
+            # If the concept does not have a line that is RXNORM with a desired TTY (IN\BN\PIN) we will place the parameters according to the
+            # following priority: if there is IN we will choose the first one we see, if there is no IN we will choose BN
+            # and if there is no IN\BN we will choose PIN
+            elif current_cid in con_no_rxnorm:
+                if row['TTY'] == 'IN' and bn_in_pin != 'IN':
+                    bn_in_pin = row['TTY']
+                    name = row['STR']
+                elif row['TTY'] == 'BN' and bn_in_pin != 'IN':
+                    bn_in_pin = row['TTY']
+                    name = row['STR']
+                elif row['TTY'] == 'PIN' and bn_in_pin not in ('IN', 'BN'):
+                    bn_in_pin = row['TTY']
+                    name = row['STR']
+
+            # If there is one row that is with SAB=DRUGBANK we would like this concept (used to clear INs that are not necessarily drugs)
+            if row['SAB'] == 'DRUGBANK':
+                had_drugbank = True
+
+    with open(RELATIONSHIP) as rel:
+        row_iter = csv.DictReader(rel, delimiter='|',fieldnames=["RXCUI1", "RXAUI1", "STYPE1", "REL", "RXCUI2", "RXAUI2", "STYPE2",
+                                                                 "RELA", "RUI", "SRUI", "SAB", "SL", "RG", "DIR",
+                                                                 "SUPPRESS", "CVF"])
+        # move over relationship rows
+        for row in tqdm(row_iter):
+            # Linking the synonyms to the INs we saved which are RXNORM\NO RXMPRM and adding them to the appropriate alias group
+            if (row['SAB'] == 'RXNORM' and row['STYPE1'] == 'CUI' and row['STYPE2'] == 'CUI' and row['RXCUI2'] in concept_id_dict_key_rxn) or\
+                    (row['STYPE1'] == 'CUI' and row['STYPE2'] == 'CUI' and row['RXCUI2'] in concept_id_dict_key_no_rxn):
+                if row['RELA'] == 'has_form' and row['RXCUI1'] in concept_dict_pin:
+                    concept_dict[row['RXCUI2']]['precise ingredient'].append(concept_dict_pin[row['RXCUI1']][0])
+                    concept_dict[row['RXCUI2']]['precise ingredient synonyms'].update(concept_dict_pin[row['RXCUI1']][1])
+                elif row['RELA'] == 'has_tradename' and row['RXCUI1'] in concept_dict_bn:
+                    concept_dict[row['RXCUI2']]['brand name'].append(concept_dict_bn[row['RXCUI1']][0])
+                    concept_dict[row['RXCUI2']]['brand name synonyms'].update(concept_dict_bn[row['RXCUI1']][1])
+
+    # Convert the set to a list so we can write the concepts to a file
+    for k in tqdm(concept_dict.keys()):
```

Comment:
dont need to change, just want you to be familiar with alternatives. Instead of hardcoding all the fields you can do:
```
for k,fields in tqdm(concept_dict.items()):
    for field, values in tqdm(fields.items()):
         concept_dict[k][field] = list(values)
```
or something of the sort